dist/tools/codespell/check.sh

```diff
@@ -41,15 +41,23 @@ CODESPELL_OPTS+=" --check-hidden"
 # "dout => doubt"
 CODESPELL_OPTS+=" --ignore-words-list=ND,nd,wan,od,dout"
 
-# Filter-out all false positive raising "disabled due to" messages.
-ERRORS=$(${CODESPELL_CMD} ${CODESPELL_OPTS} ${FILES} | grep -ve "disabled due to")
-
-if [ -n "${ERRORS}" ]
-then
-    printf "%sThere are typos in the following files:%s\n\n" "${CERROR}" "${CRESET}"
-    printf "%s\n" "${ERRORS}"
-    # TODO: return 1 when all typos are fixed
-    exit 0
+if [ "${CODESPELL_INTERACTIVE}" = "1" ]; then
+    # interactive mode
+    CODESPELL_OPTS+=" -w -i3"
+    exec ${CODESPELL_CMD} ${CODESPELL_OPTS} ${FILES}
```

Comment:
How about the following ?

```suggestion
    exec ${CODESPELL_CMD} ${CODESPELL_OPTS} ${FILES}
    exit 0
fi
```

This way no need to change the non interactive mode code.

Reply:
exec replaces the current shell, lines after are not evaluated anymore:

```
[kaspar@ng riot (add_codespell_interactive_mode)]$ cat test.sh 
#!/bin/sh

exec /bin/echo FOO
echo bar
[kaspar@ng riot (add_codespell_interactive_mode)]$ sh test.sh 
FOO
[kaspar@ng riot (add_codespell_interactive_mode)]$
```

Reply:
Then remove `exec` keyword from my suggestion ?

Reply:
What's your aim? 

> This way no need to change the non interactive mode code.

You mean, not move into the else block?

Reply:
ping @aabadie you mean to refactor this so the original code is not in an else case?

Reply:
> you mean to refactor this so the original code is not in an else case?

Yes